force-app/main/default/classes/SOQL.cls

```diff
@@ -312,50 +505,93 @@ public with sharing class SOQL {
      * @param Original Opportunity Id
      * @param List of cloned Opportunity Ids
      */
-    public static void cloneLineItems(Id origOppId, List<Id> cloneOppIds) {
+    public void cloneLineItems(Id origOppId, List<Id> cloneOppIds) {
         // Query the original Opportunity Line Items
+        List<OpportunityLineItem> originalLineItems = 
+        [SELECT Product2Id, UnitPrice, Quantity, TotalPrice, Description, PricebookEntryId 
+        FROM OpportunityLineItem 
+        WHERE OpportunityId = :origOppId];
 
         // Create a list to hold the new line items
+        List<OpportunityLineItem> newLineItems = new List<OpportunityLineItem>();
 
         // Generate a random number between 1 and 3
+        Integer randomNumber;
 
         // Loop through each cloned Opportunity Id
-
+        for(Id cloneOppId : cloneOppIds) {
             // Loop through each original line item
-
+            for(OpportunityLineItem originalLineItem : originalLineItems) {
                 // Clone the original line item and add the random number to the quantity
-
+                OpportunityLineItem newLineItem = originalLineItem.clone(false, false, false, false);
+                newLineItem.OpportunityId = cloneOppId;
+                randomNumber = Math.round(Math.random() * 2) + 1; // Generates a random number between 1 and 3
+                newLineItem.Quantity = originalLineItem.Quantity + randomNumber;
```

Comment:
Try doing this without using opportunity.clone. manually set the line item fields from the old line item